swift/Sources/Blockchains/Bitcoin.swift

```diff
@@ -12,141 +12,21 @@ public typealias TWBitcoinUnspentTransaction = TW_Bitcoin_Proto_UnspentTransacti
 public typealias TWBitcoinTransactionPlan = TW_Bitcoin_Proto_TransactionPlan
 public typealias TWBitcoinOutPoint = TW_Bitcoin_Proto_OutPoint
 
-/// Bitcoin blockchain.
-///
-/// Bitcoin-based blockchains should inherit from this class.
-open class Bitcoin: Blockchain {
-    /// SLIP-044 coin type.
-    override open var coinType: CoinType {
-        return .bitcoin
-    }
-
-    override open var xpubVersion: HDVersion? {
-        switch self.coinPurpose {
-        case .bip44:
-            return HDVersion.xpub
-        case .bip49:
-            return HDVersion.ypub
-        case .bip84:
-            return HDVersion.zpub
-        }
-    }
-
-    override open var xprvVersion: HDVersion? {
-        switch self.coinPurpose {
-        case .bip44:
-            return HDVersion.xprv
-        case .bip49:
-            return HDVersion.yprv
-        case .bip84:
-            return HDVersion.zprv
-        }
-    }
-
-    /// Private key prefix.
-    ///
-    /// - SeeAlso: https://en.bitcoin.it/wiki/List_of_address_prefixes
-    open var privateKeyPrefix: UInt8 {
-        return 0x80
-    }
-
-    /// Public key hash address prefix.
-    ///
-    /// - SeeAlso: https://en.bitcoin.it/wiki/List_of_address_prefixes
-    open var p2pkhPrefix: UInt8 {
-        return P2PKHPrefix.bitcoin.rawValue
-    }
-
-    /// Pay to script hash (P2SH) address prefix.
-    ///
-    /// - SeeAlso: https://en.bitcoin.it/wiki/List_of_address_prefixes
-    open var p2shPrefix: UInt8 {
-        return P2SHPrefix.bitcoin.rawValue
-    }
-
-    public var prefixSet: Set<UInt8> {
-        return Set([p2shPrefix, p2pkhPrefix])
-    }
-
-    open var hrp: HRP {
-        return .bitcoin
-    }
-
-    open var supportSegwit: Bool {
-        return true
-    }
-
-    override open func address(for publicKey: PublicKey) -> Address {
-        switch coinPurpose {
+public extension HDWallet {
+    static func derive(from extended: String, at path: DerivationPath) -> PublicKey? {
+        let xpubVersion: HDVersion
+        let xprvVersion: HDVersion
+        switch path.purpose {
         case .bip44:
-            let hash = Data([p2pkhPrefix]) + publicKey.bitcoinKeyHash
-            return BitcoinAddress(data: hash)!
+            xpubVersion = path.coinType == .litecoin ? .ltub: .xpub
+            xprvVersion = path.coinType == .litecoin ? .ltpv: .xprv
```

Comment:
can you add a test for this one? it is a little tricky logic

Reply:
Yes, we have ltc tests to cover this